docgenerator/src/main/scala/codacy/codenarc/docs/DocGenerator.scala

```diff
@@ -0,0 +1,316 @@
+package codacy.codenarc.docs
+
+import java.lang.reflect.Field
+
+import better.files._
+import com.codacy.plugins.api.results.{Parameter, Pattern, Result, Tool}
+import org.codenarc.rule.AbstractAstVisitorRule
+import play.api.libs.json.Json
+
+import scala.jdk.CollectionConverters._
+import org.reflections.Reflections
+
+import scala.annotation.tailrec
+import scala.util.{Try, Using}
+
+object DocGenerator {
+
+  case class RulePriority(priority: String, priorityType: String)
+
+  case class RulePatternInformation(
+      patternSpecification: Pattern.Specification,
+      patternDescription: Pattern.Description,
+      descriptionMarkdown: String
+  )
+
+  private val toolName = "codenarc"
+
+  private val codeNarcVersionFile = ".codenarc-version"
+  private val codeNarcGitRepository = "github.com/CodeNarc/CodeNarc"
+  private val patternsJsonFile = "patterns.json"
+  private val descriptionJsonFile = "description.json"
+  private val docsFolder = "docs"
+  private val descriptionFolder = "description"
+
+  private val codeNarcDocumentationStartingPoint = "codenarc-rule-index.md"
+
+  private val reflections = new Reflections("org.codenarc.rule")
+  private val subTypes = reflections.getSubTypesOf(classOf[org.codenarc.rule.AbstractAstVisitorRule]).asScala
+
+  /**
+    * Converts from CodeNarc's complexity to Codacy level
+    */
+  def levelFromPriority(priority: Option[RulePriority]): Result.Level.Value = priority match {
+    case Some(RulePriority("1", _)) => Result.Level.Err
+    case Some(RulePriority("2", _)) => Result.Level.Warn
+    case Some(RulePriority("3", _)) => Result.Level.Info
+    case _ => Result.Level.Info
+  }
+
+  def categoryTypeFromPriority(priorityType: Option[RulePriority]): Pattern.Category.Value = priorityType match {
```

Comment:
Here you pass `Option[String]` and call it `categoryTypeFromPriorityType`

Reply:
Done ✅